core/src/main/java/hudson/Util.java

```diff
@@ -568,6 +568,20 @@ public static boolean isRelativePath(String path) {
     }
 
     /**
+     * A check if a file path is a descendant of a parent path
+     * @param forParent the parent the child should be a descendant of
+     * @param potentialChild the path to check
+     * @return true if so
+     * @throws IOException potentially from {@link Files#isSameFile(Path, Path)}
+     * @since FIXME
+     */
+    public static boolean isDescendant(File forParent, File potentialChild) throws IOException {
+        Path child = potentialChild.getAbsoluteFile().toPath();
```

Comment:
Ah, it's a left over from when I was actually walking the path and checking each path down if it was actually the same on the file system. Unfortunately that required the files to actually exist on disk which didn't work for my purpose.

I'll do a canonical thing here.